timescale/timescale.go

```diff
@@ -38,6 +38,23 @@ func (r repo) RetrieveAll(ctx context.Context, pm callhome.PageMetadata) (callho
 		  LIMIT :limit OFFSET :offset
 	  ) t ON ad.ip_address = t.ip_address;	  
 	`
+	if pm.Limit == 0 {
+		q = `
+	WITH aggregated_data AS (
+		SELECT ip_address, ARRAY_AGG(DISTINCT service) AS services
+		FROM telemetry
+		GROUP BY ip_address
+	  )
+	  SELECT ad.ip_address, ad.services, t.time, t.service_time, t.longitude, t.latitude, t.mf_version, t.country, t.city
+	  FROM aggregated_data ad
+	  INNER JOIN (
+		  SELECT DISTINCT ON (ip_address) *
+		  FROM telemetry
+		  ORDER BY ip_address, time DESC
+		  OFFSET :offset
+	  ) t ON ad.ip_address = t.ip_address;	  
+	`
+	}
```

Comment:
Instead of reasigning query variable, assign query varaible once with switch case
```suggestion

switch {
    case pm.Limit == 0:
        q = `Long sql statement`
    default :
        q = `Long sql statement`
}
```